azkaban-common/src/test/java/azkaban/project/DirectoryYamlFlowLoaderTest.java

```diff
@@ -37,31 +52,136 @@ public void setUp() {
+      Assert.assertNull(edge.getError());
+    }
+
+    Assert.assertEquals(2, loader.getEdgeMap().get(BASIC_FLOW_2).size());
+    Assert.assertEquals(2, flow2.getEdges().size());
+    for (final Edge edge : loader.getEdgeMap().get(BASIC_FLOW_2)) {
+      this.logger.info(BASIC_FLOW_2 + ".flow has edge: " + edge.getId());
+      Assert.assertNull(edge.getError());
+    }
+  }
+
+  @Test
+  public void testLoadEmbeddedFlowYamlFile() {
+    final DirectoryYamlFlowLoader loader = new DirectoryYamlFlowLoader(new Props());
+
+    loader.loadProjectFlow(this.project, ExecutionsTestUtil.getFlowDir(EMBEDDED_FLOW_YAML_DIR));
+    Assert.assertEquals(0, loader.getErrors().size());
+    Assert.assertEquals(3, loader.getFlowMap().size());
+    Assert.assertTrue(loader.getFlowMap().containsKey(EMBEDDED_FLOW));
+    final Flow flow = loader.getFlowMap().get(EMBEDDED_FLOW);
+    Assert.assertEquals(4, flow.getNodes().size());
+
+    Assert.assertTrue(loader.getFlowMap().containsKey(EMBEDDED_FLOW_1));
+    final Flow flow1 = loader.getFlowMap().get(EMBEDDED_FLOW_1);
+    Assert.assertEquals(4, flow1.getNodes().size());
+
+    Assert.assertTrue(loader.getFlowMap().containsKey(EMBEDDED_FLOW_2));
+    final Flow flow2 = loader.getFlowMap().get(EMBEDDED_FLOW_2);
+    Assert.assertEquals(2, flow2.getNodes().size());
+
+    // Verify flow edges
+    Assert.assertEquals(3, loader.getEdgeMap().size());
+    Assert.assertEquals(3, loader.getEdgeMap().get(EMBEDDED_FLOW).size());
+    Assert.assertEquals(3, flow.getEdges().size());
+    for (final Edge edge : loader.getEdgeMap().get(EMBEDDED_FLOW)) {
+      this.logger
+          .info(EMBEDDED_FLOW + ".flow has edge: " + edge.getId());
+      Assert.assertNull(edge.getError());
+    }
+
+    Assert.assertEquals(3, loader.getEdgeMap().get(EMBEDDED_FLOW_1).size());
+    for (final Edge edge : loader.getEdgeMap().get(EMBEDDED_FLOW_1)) {
+      this.logger.info(
+          EMBEDDED_FLOW_1 + ".flow has edge: " + edge.getId());
+      Assert.assertNull(edge.getError());
+    }
+
+    Assert.assertEquals(1, loader.getEdgeMap().get(EMBEDDED_FLOW_2).size());
+    for (final Edge edge : loader.getEdgeMap().get(EMBEDDED_FLOW_2)) {
+      this.logger.info(EMBEDDED_FLOW_2 + ".flow has edge: " + edge.getId());
+      Assert.assertNull(edge.getError());
+    }
+  }
+
+  @Test
+  public void testLoadInvalidFlowYamlFiles() {
+    final DirectoryYamlFlowLoader loader = new DirectoryYamlFlowLoader(new Props());
+
+    loader.loadProjectFlow(this.project, ExecutionsTestUtil.getFlowDir(INVALID_FLOW_YAML_DIR));
+    Assert.assertEquals(2, loader.getErrors().size());
+    Assert.assertEquals(2, loader.getFlowMap().size());
+    Assert.assertEquals(2, loader.getEdgeMap().size());
+
+    // Invalid flow 1: Dependency not found.
+    Assert.assertTrue(loader.getFlowMap().containsKey(INVALID_FLOW_1));
+    final Flow flow1 = loader.getFlowMap().get(INVALID_FLOW_1);
+
+    Assert.assertEquals(3, flow1.getNodes().size());
```

Comment:
If I were to write this test, I would write a common assert method to consolidate all similar check statements.

Reply:
Thanks for the suggestion. I'll consolidate the common check statements.